plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3RepositoryPlugin.java

```diff
@@ -77,15 +78,17 @@ public class S3RepositoryPlugin extends Plugin implements RepositoryPlugin, Relo
     }
 
     protected final S3Service service;
+    private final Path configPath;
 
-    public S3RepositoryPlugin(final Settings settings) {
-        this(settings, new S3Service());
+    public S3RepositoryPlugin(final Settings settings, final Path configPath) {
```

Comment:
If anyone else reading this is wondering how we can change a public constructor without changing any calling code, the answer is that the PluginsService expects one of three different constructor signatures [and this one of them](https://github.com/opensearch-project/OpenSearch/blob/e41cbe52134488ab29dcf6276dc2fea65c7f5105/server/src/main/java/org/opensearch/plugins/PluginsService.java#L782).